pilot/pkg/security/trustdomain/bundle_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trustdomain
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestReplaceTrustDomainAliases(t *testing.T) {
+	testCases := []struct {
+		name              string
+		trustDomainBundle Bundle
+		principals        []string
+		expect            []string
+	}{
+		{
+			name:              "No trust domain aliases (no change in trust domain)",
+			trustDomainBundle: NewTrustDomainBundle("cluster.local", nil),
+			principals:        []string{"cluster.local/ns/foo/sa/bar"},
+			expect:            []string{"cluster.local/ns/foo/sa/bar"},
+		},
+		{
+			name:              "One trust domain alias, one principal",
+			trustDomainBundle: NewTrustDomainBundle("td2", []string{"td1"}),
+			principals:        []string{"td1/ns/foo/sa/bar"},
+			expect:            []string{"td2/ns/foo/sa/bar", "td1/ns/foo/sa/bar"},
+		},
+		{
+			name:              "One trust domain alias, two principals",
+			trustDomainBundle: NewTrustDomainBundle("td1", []string{"cluster.local"}),
+			principals:        []string{"cluster.local/ns/foo/sa/bar", "cluster.local/ns/yyy/sa/zzz"},
+			expect:            []string{"td1/ns/foo/sa/bar", "cluster.local/ns/foo/sa/bar", "td1/ns/yyy/sa/zzz", "cluster.local/ns/yyy/sa/zzz"},
+		},
+		{
+			name:              "Two trust domain aliases, two principals",
+			trustDomainBundle: NewTrustDomainBundle("td2", []string{"td1", "cluster.local"}),
+			principals:        []string{"cluster.local/ns/foo/sa/bar", "td1/ns/yyy/sa/zzz"},
+			expect: []string{"td2/ns/foo/sa/bar", "td1/ns/foo/sa/bar", "cluster.local/ns/foo/sa/bar",
+				"td2/ns/yyy/sa/zzz", "td1/ns/yyy/sa/zzz", "cluster.local/ns/yyy/sa/zzz"},
+		},
+		{
+			name:              "Principals not match alias",
+			trustDomainBundle: NewTrustDomainBundle("td1", []string{"td2"}),
+			principals:        []string{"some-td/ns/foo/sa/bar"},
+			expect:            []string{},
```

Comment:
Should we return the principals as-is if the user just want put this principal in the authz policy?

Reply:
I don't have a preference, but the requests will be rejected anyway (whether the output principals are empty or the existing one), since `some-td` doesn't match with the new td or the old td.

Reply:
@liminw

Reply:
The PR now returns principals as-is if they don't match the trust domain or aliases.

Reply:
Returning principals as-is sounds good to me.